R/zzz.R

```diff
@@ -1,8 +1,19 @@
 .onLoad <- function(libname, pkgname) {
   # adapted from https://github.com/r-lib/devtools/blob/master/R/zzz.R
+
+  lockfile_flag <-
+    !(
+      identical(Sys.getenv("CALLR_IS_RUNNING"), "true") || # inside callr process
+      identical(Sys.getenv("TESTTHAT"), "true") || # inside devtools::test
```

Comment:
a question
what would happen if we do `load_all()` and then `test()`? 
Do I understand correctly that the option would be set to true and then lockfile created inside the tests?

Reply:
Let me check, good question

Reply:
I actually think that `Sys.getenv("TESTTHAT")` flag is set after the package is loaded during tests, and the initial value of `getOption('teal.lockfile.enable')` is set to `TRUE` during tests. Not as I would expect it to work

I added this snippet to check the behavior of this flag during checks.

```r
cat('\n teal.lockfile.enable: ', getOption('teal.lockfile.enable'), '\n')
```

This made me think we actually need to disable the creation of the lockfile inside `srv_teal_lockfile`, if `Sys.getenv("TESTTHAT")` gets set later after the package is loaded

<img width="727" alt="image" src="https://github.com/user-attachments/assets/934368ee-0dfa-4337-ac1a-f42f4ca0be3a">

Reply:
`Sys.getenv("TESTTHAT")` is set at `testthat::test_that`, not at `devtools::test()`

<img width="560" alt="image" src="https://github.com/user-attachments/assets/4e113d7d-889d-4d88-bcda-64e33bd13639">

Reply:
> This made me think we actually need to disable the creation of the lockfile inside srv_teal_lockfile, if Sys.getenv("TESTTHAT") gets set later after the package is loaded

I got a similar idea. And I was thinking about this particular example in my previous comment [here](https://github.com/insightsengineering/teal/pull/1346#discussion_r1764903472)

Reply:
With this commit https://github.com/insightsengineering/teal/pull/1346/commits/18620432f1db0de5d99ca23922dd80a713615be8 I moved the functionality inside `srv_teal_lockfile` so the checks are made on lockfile execution.
If the option was set before the lockfile was called, then the option is superior to the circumstances check